### PR TITLE
SHS-NG M4.1: Port environment page to new backend.

### PR DESCRIPTION
**Pull request by vanzin**
This change moves the environment page to the new UI module,
making modifications so that its data is stored in and retrieved
from the new store.

Some temporary internal API changes were made to maintain the current
code working when using the new UI implementation. These may be cleaned
up when the old UI code is cleaned up and only the new implementation
remains.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/ui/SparkUI.scala [hunk 1]
```diff
@@ -88,9 +86,13 @@ private[spark] class SparkUI private (
   initialize()
 
   def getSparkUser: String = {
-    environmentListener.sparkUser
-      .orElse(environmentListener.systemProperties.toMap.get("user.name"))
-      .getOrElse("<unknown>")
+    try {
+      Option(store.applicationInfo().attempts.head.sparkUser)
+        .orElse(store.environmentInfo().systemProperties.toMap.get("user.name"))
```
Comment (squito) [inline core/src/main/scala/org/apache/spark/ui/SparkUI.scala:91]: just to double check -- its not possible that there would be a `store.environmentInfo()` but no `store.applicationInfo()`??

**Review by squito — COMMENTED** [review]
looks fine.  My only real concern is that there don't seem to be any tests for this in the UI -- I don't see anything covering this in UISeleniumSuite or HistoryServerSuite.  Is there anything else that covers it?

If not, I think it would be good to just add something really simple in UISeleniumSuite.

**Comment by vanzin** [comment]
@squito I don't think there are any tests other than the ones I added here, which only cover the listener part.

File: core/src/main/scala/org/apache/spark/ui/SparkUI.scala (hunk 1 above)
Comment (vanzin) [reply core/src/main/scala/org/apache/spark/ui/SparkUI.scala:91]: `SparkContext` does this (~ L556):

```
    postEnvironmentUpdate()
    postApplicationStart()
```

Comment (squito) [reply core/src/main/scala/org/apache/spark/ui/SparkUI.scala:91]: ok, so it is possible you'd have an environment but no application start -- but there wouldn't be anything else meaningful to say in this case anyway, so no need to worry about it.

**Pull request closed without merging**